plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/resource/FordiacTypeResource.java

```diff
@@ -116,21 +132,37 @@ protected void doSave(final OutputStream outputStream, final Map<?, ?> options)
 					MessageFormat.format(Messages.FordiacTypeResource_SaveToUnsupportedURI, uri.toString()));
 		}
 		try {
-			final var fbtFile = ResourcesPlugin.getWorkspace().getRoot()
-					.getFile(new Path(this.uri.toPlatformString(true)));
-			var typeEntryForFile = TypeLibraryManager.INSTANCE.getTypeEntryForFile(fbtFile);
-			if (typeEntryForFile == null) {
-				typeEntryForFile = TypeLibraryManager.INSTANCE.getTypeLibrary(fbtFile.getProject())
-						.createTypeEntry(fbtFile);
-			}
-			typeEntryForFile.setTypeEditable(EcoreUtil.copy((LibraryElement) getContents().get(0)));
-			typeEntryForFile.setLastModificationTimestamp(typeEntryForFile.getFile().getModificationStamp());
-			AbstractTypeExporter.saveType(typeEntryForFile, outputStream);
+			saveType(getContents().get(0), outputStream);
 		} catch (final Exception e) {
 			throw new IOWrappedException(e);
 		}
 	}
 
+	public static void saveType(final EObject le, final OutputStream outputStream) {
```

Comment:
what is le ?
i think libraryElement would be better.
I am not a big fan of short variable names, but that is a matter of taste

Reply:
yes you are right. will fix it right after.